Code/Framework/AzToolsFramework/AzToolsFramework/Manipulators/SurfaceManipulator.cpp

```diff
@@ -95,17 +80,29 @@ namespace AzToolsFramework
     }
 
     void SurfaceManipulator::OnLeftMouseDownImpl(
-        const ViewportInteraction::MouseInteraction& interaction, float /*rayIntersectionDistance*/)
+        const ViewportInteraction::MouseInteraction& interaction, [[maybe_unused]] float rayIntersectionDistance)
     {
         const AZ::Transform worldFromLocalUniformScale = TransformUniformScale(GetSpace());
 
-        const GridSnapParameters gridSnapParams = GridSnapSettings(interaction.m_interactionId.m_viewportId);
+        const AzFramework::ViewportId viewportId = interaction.m_interactionId.m_viewportId;
+
+        m_rayRequest.m_entityFilter.m_ignoreEntities.clear();
+        const auto& entityComponentIdPairs = EntityComponentIdPairs();
+        AZStd::transform(
```

Comment:
would it be worth to call reserve() before the insertions?

Reply:
Yeah good shout I can add that :+1: